agent/connect/ca/provider_vault.go

```diff
@@ -172,7 +172,7 @@ func (v *VaultProvider) GenerateIntermediate() (string, error) {
 			"allow_any_name":   true,
 			"allowed_uri_sans": "spiffe://*",
 			"key_type":         "any",
-			"max_ttl":          "72h",
+			"max_ttl":          fmt.Sprintf("%.0fm", v.config.LeafCertTTL.Minutes()),
```

Comment:
won't `v.config.LeafCertTTL.String()` give you a valid duration?

Reply:
good catch! I was overcomplicating this for some reason